pages/settings/teams.tsx

```diff
@@ -1,46 +1,59 @@
+import { GetServerSideProps } from "next";
 import Head from 'next/head';
-import prisma from '../../lib/prisma';
-import Modal from '../../components/Modal';
 import Shell from '../../components/Shell';
 import SettingsShell from '../../components/Settings';
 import { useEffect, useState } from 'react';
+import type { Session } from "next-auth";
 import { useSession, getSession } from 'next-auth/client';
 import {
   UsersIcon,
 } from "@heroicons/react/outline";
 import TeamList from "../../components/team/TeamList";
 import TeamListItem from "../../components/team/TeamListItem";
 
-export default function Teams(props) {
-
+export default function Teams() {
   const [session, loading] = useSession();
   const [teams, setTeams] = useState([]);
   const [invites, setInvites] = useState([]);
   const [showCreateTeamModal, setShowCreateTeamModal] = useState(false);
 
-  const loadTeams = () => fetch('/api/user/membership').then((res: any) => res.json()).then(
-    (data) => {
+  const handleErrors = async (resp) => {
```

Comment:
getServerSideProps can be removed entirely if you add:

```javascript
if (!resp.ok) {
  // if response status === 403
  // useRouter()
  // return router.push('/auth/login');
  }
}
```

But I'll approve, as I don't consider optimizations like this blocking.